### PR TITLE
add only approved items filter to table

### PR DESCRIPTION
**Pull request by hobbitronics**
- added a filter and alternate button label for Approved Items

image

image

### CODE REVIEW

File: src/pages/policies/[policyId].svelte [hunk 1]
```diff
@@ -226,7 +227,9 @@ th {
 
   <div class="flex justify-between align-items-center">
     <h4>Items <span class="subtext">({approvedItems?.length} covered)</span></h4>
-    <Button disabled={allItemsBtnDisabled} on:click={() => (showAllItems = true)}>All Items…</Button>
+    <Button disabled={allItemsBtnDisabled} on:click={() => (showAllItems = true, showApprovedOnly = false)}>all items…</Button>
+
+    <Button on:click={() => (showApprovedOnly = !showApprovedOnly, showAllItems = false)}>toggle approved only</Button>
```
Comment (forevermatt) [inline src/pages/policies/[policyId].svelte:232]: It seems like it would be clearer to extract these assignments to simple, clearly named functions. Having multiple assignments inline like this was a little hard to parse (mentally).

File: src/pages/policies/[policyId].svelte [hunk 2]
```diff
@@ -48,7 +49,7 @@ $: members = policy.members || []
 
 $: policyId && loadItems(policyId)
 // sort items so inactive is last
-$: items = $selectedPolicyItems.filter(itemIsActive)
+$: items = $selectedPolicyItems.filter(showApprovedOnly ? itemIsApproved : itemIsActive)
```
[the remark this answers is not in the document]
Comment (hobbitronics) [reply src/pages/policies/[policyId].svelte:52]: Active is just excludes `ItemCoverageStatus.Inactive`

[the remark this answers is not in the document]
Comment (forevermatt) [reply src/pages/policies/[policyId].svelte:52]: Ahh, thanks. To clarify what I was getting at (see [this Trello card](https://trello.com/c/eIkktWRb/277-item-status-label-approved-%E2%86%92-active)), we want to eventually show the term "Active" to end users for any `Approved` items, which is how I came to the conclusion that "Active" and "Approved" are synonymous.

In a manner of speaking, everything that isn't `Approved` is inactive (not just the `Inactive` state). We're using the state `Inactive` to basically refer to deleted items, but without actually deleting the record from the database. As weird as it sounds, it might work to change `itemIsActive` here to `itemIsNotInactive` (since the word "Inactive" has that special meaning in this app).

@merchako, this is what I was thinking of that I mentioned in our meeting today.

[the remark this answers is not in the document]
Comment (hobbitronics) [reply src/pages/policies/[policyId].svelte]: So now there is one filter button that either shows all items or Approved Items limited to 15 in the ItemsTable.

**Review by forevermatt — APPROVED** [review]
